modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/UniqueTokenFilter.java

```diff
@@ -49,33 +49,29 @@
     }
 
     @Override
-    public final boolean incrementToken() throws IOException {
-        while (input.incrementToken()) {
-            final char term[] = termAttribute.buffer();
-            final int length = termAttribute.length();
+    protected boolean accept() {
+        final char term[] = termAttribute.buffer();
+        final int length = termAttribute.length();
 
-            boolean duplicate;
-            if (onlyOnSamePosition) {
-                final int posIncrement = posIncAttribute.getPositionIncrement();
-                if (posIncrement > 0) {
-                    previous.clear();
-                }
+        boolean duplicate;
+        final int posIncrement = posIncAttribute.getPositionIncrement();
 
-                duplicate = (posIncrement == 0 && previous.contains(term, 0, length));
-            } else {
-                duplicate = previous.contains(term, 0, length);
+        if (onlyOnSamePosition) {
+            if (posIncrement > 0) {
+                previous.clear();
             }
 
-            // clone the term, and add to the set of seen terms.
-            char saved[] = new char[length];
-            System.arraycopy(term, 0, saved, 0, length);
-            previous.add(saved);
-
-            if (!duplicate) {
-                return true;
-            }
+            duplicate = (posIncrement == 0 && previous.contains(term, 0, length));
+        } else {
+            duplicate = previous.contains(term, 0, length);
         }
-        return false;
+
+        // clone the term, and add to the set of seen terms.
+        char saved[] = new char[length];
+        System.arraycopy(term, 0, saved, 0, length);
+        previous.add(saved);
+
+        return !duplicate;
```

Comment:
Can you write this as `duplicate == false`?  We prefer this style as it's easy to miss prefixed `!` characters.